File: heap/bloco.cpp
#include "bloco.h"
#include<iostream>
Bloco::Bloco(void *informacao, int chave){
    this->setInformacao(informacao);
    this->setChave(chave);
}

void Bloco::setInformacao(void *informacao){
    if(informacao != NULL)
        this->informacao = informacao;
}

void Bloco::setChave(int chave){
    this->chave = chave;
}

int Bloco::getChave(){
    return this->chave;
}

bool Bloco::operator >(Bloco bloco){
    return this->getChave() > bloco.getChave() ? true :  false;
}

File: Heap/src/Bloco.java
public class Bloco {
	private Object informacao;
	private int prioridade;
	
	public Bloco(Object object, int prioridade) {
		this.setInformacao(object);
		this.setPrioridade(prioridade);
	}
	
	public void setPrioridade(int prioridade) {
		this.prioridade = prioridade > 0 ? prioridade : 0;
	}
	
	public Object getInformacao() {
		return informacao;
	}
	
	public int getPrioridade() {
		return prioridade;
	}
	
	public void setInformacao(Object informacao) {
		if(informacao != null)
			this.informacao = informacao;
	}
	
}
